File: JAVA PROGRAMS/SEM2_LAB/TowerOfHanoi.java
package SEM2_LAB;

public class TowerOfHanoi {

    static void TOH(int n,char from, char to, char help){
        if(n==0){
            return;
        }

        TOH(n-1,from,help,to);
        System.out.println("Move "+n+" from "+from+" to "+to);
        TOH(n-1,help,to,from);

    }

    public static void main(String[] args) {
        TOH(2, 'A', 'B', 'C');
    }
}

File: C PROGRAMS/DAA 3rd Sem/TowerOfHanoi.c
#include <stdio.h>

void towerOfHanoi(int n, char from, char to, char help){
    if(n==1){
        printf("Move disk from %c to %c\n", from, to);
        return;
    }
    
    towerOfHanoi(n-1, from, help, to);
    printf("Move disk from %c to %c\n", from, to);
    towerOfHanoi(n-1, help, to, from);
}

int main() {
    
    int n;
    printf("Enter number of disks: ");
    scanf("%d",&n);
    
    towerOfHanoi(n, 'A', 'C', 'B');

    return 0;
}